crates/pgt_workspace/src/workspace/server.rs

```diff
@@ -253,6 +259,88 @@ impl Workspace for WorkspaceServer {
         Ok(self.is_ignored(params.pgt_path.as_path()))
     }
 
+    fn pull_code_actions(
+        &self,
+        params: code_actions::CodeActionsParams,
+    ) -> Result<code_actions::CodeActionsResult, WorkspaceError> {
+        let doc = self
+            .documents
+            .get(&params.path)
+            .ok_or(WorkspaceError::not_found())?;
+
+        let eligible_statements = doc
+            .iter_statements_with_text_and_range()
+            .filter(|(_, range, _)| range.contains(params.cursor_position));
+
```

Comment:
we should check that the statement is valid by checking if there is a pg_query AST for it

Reply:
great point, done.